### PR TITLE
[Product] Remove ProductBuilder class

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets | #3113 
| License       | MIT
| Doc PR        | -

Rebase of #3288

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ This chapter references the relevant changes done in 0.16 version.
 To get the diff between two versions, go to https://github.com/Sylius/Sylius/compare/v0.15.0...master
 
  * feature #3110 [BC BREAK] Bumped minimal versions, major changes: PHP >=5.5.9, Symfony ^2.7
+ * feature #3288 [BC BREAK][Product] Removed ProductBuilder class
```
Comment (michalmarcinkowski) [inline CHANGELOG.md]: Entries in the CHANGELOG should be order by merge time not PR number. On the very top we should have the latest change that landed in the master. Can you make a separate PR and reorder this entries?

**Comment by michalmarcinkowski** [comment]
Thank you Janek!

**Comment by pjedrzejewski** [comment]
:+1:
